pkg/cosign/tuf/client.go

```diff
@@ -50,8 +50,8 @@ const (
 var (
 	// singletonTUF holds a single instance of TUF that will get reused on
 	// subsequent invocations of initializeTUF
-	singletonTUF   *TUF
-	singletonTUFMu sync.Mutex
+	singletonTUF     *TUF
+	singletonTUFOnce = new(sync.Once)
```

Comment:
Wow, I actually didn't know that go had allocation via `new`. 

I don't mind this approach, but I found that someone else had the same problem and made sync but with Reset - https://github.com/matryer/resync

Reply:
yeah i saw this as well and ideally wanted to avoid a totally new dependency just for testing -- I'm not sure if there's a big downside to this approach though.